src/db.rs

```diff
@@ -479,11 +479,28 @@ pub(crate) fn alias<C: ToSql>(
     Ok(())
 }
 
-pub(crate) fn reverse_alias(txn: &Transaction, cid: impl ToSql) -> crate::Result<Vec<Vec<u8>>> {
-    let id = get_id(txn, cid)?;
+pub(crate) fn resolve<C: FromSql>(txn: &Transaction, name: &[u8]) -> crate::Result<Option<C>> {
     Ok(txn
         .prepare_cached(
             r#"
+SELECT cids.cid FROM aliases JOIN cids ON aliases.block_id = cids.id AND aliases.name = ?
+"#,
+        )?
+        .query_row(params![name], |row| row.get(0))
+        .optional()?)
+}
+
+pub(crate) fn reverse_alias(
+    txn: &Transaction,
+    cid: impl ToSql,
+) -> crate::Result<Option<Vec<Vec<u8>>>> {
+    let id = get_id(txn, cid)?;
```

Comment:
Maybe it is my years with scala and fp, but I would have written this as an expression instead of using an early return.

Something like

```
Ok(if let Some(id) = ... {} else {})
```

But this is perfectly fine rust code, just a matter of taste.

Reply:
good point, in this case I think that would indeed be better style